### DISCUSSION TITLE
How to fork a readable stream

### DISCUSSION THREAD

**Discussion by kunalgoyal9** [denoland/deno#16067] [Q&A] [answered]
I need something like this

`const rStream = <get a readable stream>`
`const fStream = <get a writable stream to a file handle>`
`  rStream.on.("data", d => {`
   ` <add to sha logic>`
    `<copy to fStream>`
`})`

`rStream` is a `Deno.Reader` object. Another way I can think of is to copy this `rStream`. But making a clone with `structuredClone` doesn't work as well

**Comment by zachauten** [comment, marked as answer]
`Deno.Reader` isn't a `ReadableStream`, but you can convert it with [readableStreamFromReader](https://deno.land/std@0.157.0/streams/conversion.ts?s=readableStreamFromReader) from the std lib. Then you can use [ReadableStream.tee](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/tee) to read from it twice.
